### DISCUSSION TITLE
Removing row number parent from export.

### DISCUSSION THREAD

**Discussion by AndrewGalea** [Synthoid/ExportSheetData#160] [Q&A] [answered]
When I export as JSON, I get the following output:

`{
  "1": {
    "Pos": 1,
    "Player": "Robby",
    "God": "Grul",
    "Win": 4,
    "Loss": 0,
    "Souls": 20,
    "Levels": 41,
    "Rankings Pts": 50
  },
  "2": {
    "Pos": 2,
    "Player": "Molga",
    "God": "Torin",
    "Win": 3,
    "Loss": 1,
    "Souls": 17,
    "Levels": 44,
    "Rankings Pts": 45.83
  }
}`

I don't want the row number parent; however, I cannot work out how to do that. I mucked around with the `Nested Elements` feature, which sent me down a rabbit hole. :)

**Comment by Synthoid** [comment]
Can you post your original sheet (or just a screenshot) so I can see how it is formatted? Please note any settings you have enabled for the export too.

**Reply by AndrewGalea** [reply]
![rankings](https://github.com/Synthoid/ExportSheetData/assets/15220434/1757be5f-5340-45b3-90ea-e578d4c9379a)

**Reply by AndrewGalea** [reply]
{
  "exportType": "jsonFormat",
  "exportFolderType": "default",
  "exportFolder": "",
  "exportSheets": "currentSheet",
  "targetSheets": "{}",
  "replaceExistingFiles": true,
  "unwrapSingleRows": false,
  "collapseSingleRows": false,
  "ignoreEmptyCells": false,
  "ignoreColumnsWithPrefix": true,
  "minifyData": false,
  "includeFirstColumn": true,
  "ignorePrefix": "NOEX_",
  "unwrapSheetsWithPrefix": false,
  "unwrapPrefix": "US_",
  "collapseSheetsWithPrefix": false,
  "collapsePrefix": "CS_",
  "nestedElements": false,
  "forceArrayNest": false,
  "forceArrayPrefixNest": "NA_",
  "forceString": false,
  "exportCellArray": false,
  "exportSheetArray": false,
  "exportValueArray": false,
  "exportContentsAsArray": false,
  "exportCellObject": false,
  "emptyValueFormat": "null",
  "nullValueFormat": "null",
  "separatorChar": ",",
  "forceArray": false,
  "forceArrayPrefix": "JA_",
  "exportChildElements": false,
  "exportBoolsAsInts": false,
  "rootElement": "data",
  "nameReplacementChar": "_",
  "includeDeclaration": false,
  "declarationVersion": "1.0",
  "declarationEncoding": "none",
  "declarationStandalone": "none",
  "forceAttributes": false,
  "attributePrefix": "ATT_",
  "forceChildElements": false,
  "childElementPrefix": "CE_",
  "forceInnerText": false,
  "innerTextPrefix": "IT_",
  "rootNamespace": "",
  "namespaces": []
}

**Comment by Synthoid** [comment]
Are you wanting something like this?
```json
{
  "Players": [
    {
      "Pos": 1,
      "Player": "Player01",
      "God": "Grul",
      "Win": 4,
      "Loss": 0,
      "Souls": 20,
      "Levels": 41,
      "Rankings": 50
    },
    {
      "Pos": 2,
      "Player": "Player02",
      "God": "Bruelin",
      "Win": 3,
      "Loss": 1,
      "Souls": 17,
      "Levels": 44,
      "Rankings": 45.83
    },
    {
      "Pos": 3,
      "Player": "Player03",
      "God": "Torin",
      "Win": 3,
      "Loss": 1,
      "Souls": 17,
      "Levels": 44,
      "Rankings": 45.83
    }
  ]
}
```

**Reply by AndrewGalea** [reply]
Yes; however, ideally, I would like to have the tournament name as the parent. See below for the "NOVA 2024" example.

```
{
  "NOVA 2024": [
    {
      "Pos": 1,
      "Player": "Player01",
      "God": "Grul",
      "Win": 4,
      "Loss": 0,
      "Souls": 20,
      "Levels": 41,
      "Rankings": 50
    },
    {
      "Pos": 2,
      "Player": "Player02",
      "God": "Bruelin",
      "Win": 3,
      "Loss": 1,
      "Souls": 17,
      "Levels": 44,
      "Rankings": 45.83
    },
    {
      "Pos": 3,
      "Player": "Player03",
      "God": "Torin",
      "Win": 3,
      "Loss": 1,
      "Souls": 17,
      "Levels": 44,
      "Rankings": 45.83
    }
  ]
}
```

**Comment by Synthoid** [comment]
Hmm, would this be acceptable?
```json
{
  "Tournaments": [
    {
      "id": "Tourney 01",
      "Players": [
        {
          "Pos": 1,
          "Player": "Player01",
          "God": "Grul",
          "Win": 4,
          "Loss": 0,
          "Souls": 20,
          "Levels": 41,
          "Rankings": 50
        },
        {
          "Pos": 2,
          "Player": "Player02",
          "God": "Bruelin",
          "Win": 3,
          "Loss": 1,
          "Souls": 17,
          "Levels": 44,
          "Rankings": 45.83
        },
        {
          "Pos": 3,
          "Player": "Player03",
          "God": "Torin",
          "Win": 3,
          "Loss": 1,
          "Souls": 17,
          "Levels": 44,
          "Rankings": 45.83
        }
      ]
    },
    {
      "id": "Tourney 02",
      "Players": [
        {
          "Pos": 1,
          "Player": "Player04",
          "God": "Ista",
          "Win": 3,
          "Loss": 1,
          "Souls": 17,
          "Levels": 43,
          "Rankings": 37.5
        },
        {
          "Pos": 2,
          "Player": "Player05",
          "God": "Krognar",
          "Win": 2,
          "Loss": 2,
          "Souls": 17,
          "Levels": 42,
          "Rankings": 33.33
        },
        {
          "Pos": 3,
          "Player": "Player06",
          "God": "Tomas",
          "Win": 2,
          "Loss": 2,
          "Souls": 17,
          "Levels": 39,
          "Rankings": 29.17
        }
      ]
    },
    {
      "id": "Tourney 03",
      "Players": [
        {
          "Pos": 1,
          "Player": "Player07",
          "God": "Bruelin",
          "Win": 2,
          "Loss": 2,
          "Souls": 14,
          "Levels": 42,
          "Rankings": 25
        },
        {
          "Pos": 2,
          "Player": "Player08",
          "God": "Tomas",
          "Win": 2,
          "Loss": 2,
          "Souls": 14,
          "Levels": 41,
          "Rankings": 20.83
        },
        {
          "Pos": 3,
          "Player": "Player09",
          "God": "Grul",
          "Win": 2,
          "Loss": 2,
          "Souls": 14,
          "Levels": 36,
          "Rankings": 16.67
        }
      ]
    }
  ]
}
```

**Comment by Synthoid** [comment]
Alternatively, you can do this:
```json
{
  "Tournament 01": [
    {
      "Pos": 1,
      "Player": "Player01",
      "God": "Grul",
      "Win": 4,
      "Loss": 0,
      "Souls": 20,
      "Levels": 41,
      "Rankings": 50
    },
    {
      "Pos": 2,
      "Player": "Player02",
      "God": "Bruelin",
      "Win": 3,
      "Loss": 1,
      "Souls": 17,
      "Levels": 44,
      "Rankings": 45.83
    },
    {
      "Pos": 3,
      "Player": "Player03",
      "God": "Torin",
      "Win": 3,
      "Loss": 1,
      "Souls": 17,
      "Levels": 44,
      "Rankings": 45.83
    }
  ],
  "Tournament 02": [
    {
      "Pos": 1,
      "Player": "Player04",
      "God": "Ista",
      "Win": 3,
      "Loss": 1,
      "Souls": 17,
      "Levels": 43,
      "Rankings": 37.5
    },
    {
      "Pos": 2,
      "Player": "Player05",
      "God": "Krognar",
      "Win": 2,
      "Loss": 2,
      "Souls": 17,
      "Levels": 42,
      "Rankings": 33.33
    },
    {
      "Pos": 3,
      "Player": "Player06",
      "God": "Tomas",
      "Win": 2,
      "Loss": 2,
      "Souls": 17,
      "Levels": 39,
      "Rankings": 29.17
    }
  ],
  "Tournament 03": [
    {
      "Pos": 1,
      "Player": "Player07",
      "God": "Bruelin",
      "Win": 2,
      "Loss": 2,
      "Souls": 14,
      "Levels": 42,
      "Rankings": 25
    },
    {
      "Pos": 2,
      "Player": "Player08",
      "God": "Tomas",
      "Win": 2,
      "Loss": 2,
      "Souls": 14,
      "Levels": 41,
      "Rankings": 20.83
    },
    {
      "Pos": 3,
      "Player": "Player09",
      "God": "Grul",
      "Win": 2,
      "Loss": 2,
      "Souls": 14,
      "Levels": 36,
      "Rankings": 16.67
    }
  ]
}
```

**Comment by Synthoid** [comment]
The first example keeps everything in a single sheet and uses nested elements:
![image](https://github.com/Synthoid/ExportSheetData/assets/6645950/1854fc45-c4d8-475d-9d18-3d8d2f4d503a)

The second example using a specific sheet for each tournament array:
![image](https://github.com/Synthoid/ExportSheetData/assets/6645950/5dab262c-adef-41d8-80eb-eae62e6f082f)

I'd recommend the second approach unless you really want to keep everything in one sheet. Splitting data up allows you to more conveniently group the data in an intuitive way and doesn't require advanced knowledge of nested elements.

**Reply by AndrewGalea** [reply]
I do currently have each tournament in its own tab and that is how I want to do it. What are the settings to make that happen?

**Comment by AndrewGalea** [comment]
Ok, so if I click on "Export Contents as an Array", I achieve my original goal.

[
  {
    "Pos": 1,
    "Player": "Robby",
    "God": "Grul",
    "Win": 4,
    "Loss": 0,
    "Souls": 20,
    "Levels": 41,
    "Rankings Pts": 50
  },
  {
    "Pos": 2,
    "Player": "Aaron Franklin",
    "God": "Bruelin",
    "Win": 3,
    "Loss": 1,
    "Souls": 17,
    "Levels": 44,
    "Rankings Pts": 45.83
  },
  {
    "Pos": 2,
    "Player": "Molga",
    "God": "Torin",
    "Win": 3,
    "Loss": 1,
    "Souls": 17,
    "Levels": 44,
    "Rankings Pts": 45.83
  },
  {
    "Pos": 4,
    "Player": "Guns",
    "God": "Ista",
    "Win": 3,
    "Loss": 1,
    "Souls": 17,
    "Levels": 43,
    "Rankings Pts": 37.5
  },
  {
    "Pos": 5,
    "Player": "Tirrith",
    "God": "Krognar",
    "Win": 2,
    "Loss": 2,
    "Souls": 17,
    "Levels": 42,
    "Rankings Pts": 33.33
  },
  {
    "Pos": 6,
    "Player": "Noahscape",
    "God": "Tomas",
    "Win": 2,
    "Loss": 2,
    "Souls": 17,
    "Levels": 39,
    "Rankings Pts": 29.17
  },
  {
    "Pos": 7,
    "Player": "TheConstantWay",
    "God": "Bruelin",
    "Win": 2,
    "Loss": 2,
    "Souls": 14,
    "Levels": 42,
    "Rankings Pts": 25
  },
  {
    "Pos": 8,
    "Player": "Corey",
    "God": "Tomas",
    "Win": 2,
    "Loss": 2,
    "Souls": 14,
    "Levels": 41,
    "Rankings Pts": 20.83
  },
  {
    "Pos": 9,
    "Player": "TonyJ",
    "God": "Grul",
    "Win": 2,
    "Loss": 2,
    "Souls": 14,
    "Levels": 36,
    "Rankings Pts": 16.67
  },
  {
    "Pos": 10,
    "Player": "Stephanie Berry",
    "God": "Tomas",
    "Win": 2,
    "Loss": 2,
    "Souls": 13,
    "Levels": 40,
    "Rankings Pts": 12.5
  },
  {
    "Pos": 11,
    "Player": "Kunerbus",
    "God": "Torin",
    "Win": 1,
    "Loss": 3,
    "Souls": 12,
    "Levels": 38,
    "Rankings Pts": 8.33
  },
  {
    "Pos": 12,
    "Player": "Swampmist",
    "God": "Tomas",
    "Win": 1,
    "Loss": 3,
    "Souls": 10,
    "Levels": 35,
    "Rankings Pts": 4.17
  }
]

All that is left to do is get the sheet name as a parent for the entire array.

**Comment by Synthoid** [comment, marked as answer]
You should just need to enable [Export Sheet Arrays](https://github.com/Synthoid/ExportSheetData/blob/master/docs/json/exportsheetarrays.md) to get the second result.

**Reply by AndrewGalea** [reply]
Thanks for all your help. It was appreciated.

**Reply by Synthoid** [reply]
Glad we got it resolved. Happy to help!
